internal/bundle/parse.go

```diff
@@ -0,0 +1,73 @@
+package bundle
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/model-distribution/types"
+)
+
+// Parse returns the Bundle at the given rootDir
+func Parse(rootDir string) (*Bundle, error) {
+	if fi, err := os.Stat(rootDir); err != nil || !fi.IsDir() {
+		return nil, fmt.Errorf("inspect bundle root dir: %w", err)
+	}
+	ggufPath, err := findGGUFFile(rootDir)
+	if err != nil {
+		return nil, err
+	}
+	mmprojPath, err := findMultiModalProjectorFile(rootDir)
+	if err != nil {
+		return nil, err
+	}
+	cfg, err := parseRuntimeConfig(rootDir)
+	if err != nil {
+		return nil, err
+	}
+	return &Bundle{
+		dir:           rootDir,
+		mmprojPath:    mmprojPath,
+		ggufFile:      ggufPath,
+		runtimeConfig: cfg,
+	}, nil
+}
+
+func parseRuntimeConfig(rootDir string) (types.Config, error) {
+	f, err := os.Open(filepath.Join(rootDir, "config.json"))
+	if err != nil {
+		return types.Config{}, fmt.Errorf("open runtime config: %w", err)
+	}
+	defer f.Close()
+	var cfg types.Config
+	if err := json.NewDecoder(f).Decode(&cfg); err != nil {
+		return types.Config{}, fmt.Errorf("decode runtime config: %w", err)
+	}
+	return cfg, nil
+}
+
+func findGGUFFile(rootDir string) (string, error) {
+	ggufs, err := filepath.Glob(filepath.Join(rootDir, "*.gguf"))
```

Comment:
I worry about this picking up `._` files on macOS if someone opens the directory in Finder.  Could we exclude all hidden files from the glob with something like `"[^.]*.gguf"`? (I'm not 100% sure I have the syntax right).  Same for below with `*.mmproj`.  Finder is just the worst in this regard.

Reply:
this is a really good point. I'll push a fix which will hopefully save us from an annoying debugging session in the future 😄